pkg/pillar/hypervisor/containerd.go

```diff
@@ -187,15 +185,11 @@ func (ctx ctrdContext) GetDomsCPUMem() (map[string]types.DomainMetric, error) {
 		var usedMemPerc float64
 		var cpuTotal uint64
 
-		if cg, err := cgroups.Load(cgroups.V1, cgroups.StaticPath("/eve-user-apps/"+id)); err == nil {
-			if s, err := cg.Stat(); err == nil {
-				usedMem = uint32(roundFromBytesToMbytes(s.Memory.Usage.Usage))
-				availMem = uint32(roundFromBytesToMbytes(s.Memory.Usage.Max))
-				usedMemPerc = float64(100 * float32(usedMem) / float32(availMem))
-				cpuTotal = s.CPU.Usage.Total
-			} else {
-				log.Errorf("GetDomsCPUMem failed with error %v", err)
-			}
+		if metric, err := containerd.GetMetrics(id); err == nil {
+			usedMem = uint32(roundFromBytesToMbytes(metric.Memory.Usage.Usage))
+			availMem = uint32(roundFromBytesToMbytes(metric.Memory.Usage.Max))
+			usedMemPerc = float64(100 * float32(usedMem) / float32(availMem))
```

Comment:
Could availMem be zero e.g., if we haven't started or are in the process of stopping the container?

Reply:
I don't think there's any chance for it to be 0 since the only way we get inside this block is when a container has a live task associated with it (now the task may be stopped -- but the PID exists and consumes resources at that point).

that said -- lets do a bit of defensive programming here. Although looking at gopsutils it seems they are relying on similar statement what I mentioned above: https://github.com/lf-edge/eve/blob/39bc2059080f3ee34a23c2a02448130c9eb7ba52/pkg/pillar/vendor/github.com/shirou/gopsutil/process/process.go#L236